resources/schema.py

```diff
@@ -510,6 +510,49 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         return CreatePierMutation(pier=pier)
 
 
+class UpdatePierMutation(graphene.ClientIDMutation):
+    class Input(PierInput):
+        id = graphene.ID(required=True)
+
+    pier = graphene.Field(PierNode)
+
+    @classmethod
+    @login_required
+    @superuser_required
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        # TODO: Should check if the user has permissions to
+        # delete the specific resource
+        id = from_global_id(kwargs.pop("id"))[1]
+
+        try:
+            harbor_id = kwargs.pop("harbor_id", None)
+            if harbor_id:
+                global_harbor_id = from_global_id(harbor_id)[1]
+                kwargs["harbor"] = Harbor.objects.get(pk=global_harbor_id)
+
+            pier = Pier.objects.get(pk=id)
+        except (Pier.DoesNotExist, Harbor.DoesNotExist,) as e:
+            raise VenepaikkaGraphQLError(e)
+
+        boat_types = set([])
+        for boat_type_id in kwargs.pop("suitable_boat_types", []):
+            try:
+                boat_type = BoatType.objects.get(pk=boat_type_id)
+            except BoatType.DoesNotExist as e:
+                raise VenepaikkaGraphQLError(e)
+            boat_types.add(boat_type)
+
+        try:
+            update_object(pier, kwargs)
+            pier.suitable_boat_types.set(boat_types)
+            pier.save()
```

Comment:
I think `save()` is not needed after changing m2m relations, as they are saved immediately.